src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -203,20 +203,14 @@ def __init__(
         self._retry_on_blocked = retry_on_blocked
 
         if configure_logging:
-            handler = logging.StreamHandler()
-            handler.setFormatter(CrawleeLogFormatter())
-
             root_logger = logging.getLogger()
-
-            for old_handler in root_logger.handlers[:]:
-                root_logger.removeHandler(old_handler)
-
-            root_logger.addHandler(handler)
-            root_logger.setLevel(logging.INFO if not sys.flags.dev_mode else logging.DEBUG)
+            configure_logger(root_logger, self._configuration, remove_old_handlers=True)
 
             # Silence HTTPX logger
             httpx_logger = logging.getLogger('httpx')
-            httpx_logger.setLevel(logging.WARNING if not sys.flags.dev_mode else logging.INFO)
+            httpx_logger.setLevel(
+                logging.DEBUG if get_configured_log_level(self._configuration) <= logging.DEBUG else logging.WARNING
+            )
```

Comment:
Not so important, but why did you change this and how does it differ from the previous version?

Reply:
It uses a helper function to determine the log level instead of repeating an incomplete condition all over the place. In the default case, it should behave the same as before.